### ISSUE TITLE
Make workspace context creation/disposal free threaded

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn#51016]
The .NET Project System currently switches to the UI thread for invocations of:

- [ ] `IWorkspaceProjectContextFactory.CreateProjectContext` (PR https://github.com/dotnet/roslyn/pull/51138)
- [x] `IWorkspaceProjectContext.Dispose` (PR https://github.com/dotnet/project-system/pull/6953)

https://github.com/dotnet/project-system/issues/353 is tracking updating our code to avoid those switches once these two APIs are free threaded. There did not appear to be a Roslyn issue tracking this work, hence this new issue.

**Comment by CyrusNajmabadi** [comment]
I poked around and tehse APIs already seem to be mostly free threaded (though i haven't guaranteed 100%).  Parts that are currently suspect are:

```c#
            if (languageName == LanguageNames.FSharp)
            {
                var shell = (IVsShell)ServiceProvider.GlobalProvider.GetService(typeof(SVsShell));

                // Force the F# package to load; this is necessary because the F# package listens to WorkspaceChanged to 
                // set up some items, and the F# project system doesn't guarantee that the F# package has been loaded itself
                // so we're caught in the middle doing this.
                shell.LoadPackage(Guids.FSharpPackageId, out _);
            }
```

and

```
**            // HACK: since we're on the UI thread, ensure we initialize our options provider which depends on a UI-affinitized experimentation service
            _visualStudioWorkspaceImpl.EnsureDocumentOptionProvidersInitialized();**
```

Note: instead of being synchronous-free-threaded, could we potentially just make these both async?  Would that fit into your side of things @drewnoakes ?  That woudl allow us to trivially switch over, handlign UI concerns on our end.  We coudl then move off of those pieces when we're able to, while unblocking you to not need to be on the ui thread.

Thoughts?

**Comment by drewnoakes** [comment]
@CyrusNajmabadi that seems like a good approach to me. Both our call sites are already async, so the change would be straightforward from the Project System side.

**Comment by jasonmalinowski** [comment]
@drewnoakes: Dispose() is already free-threaded, you don't need to transition for that one.

@CyrusNajmabadi: indeed, the create is _not_ free threaded, but it's all silly crap that doesn't the UI thread. Switching it to JTF-friendly async we could do; some of that also might just be fixable directly to keep it off the UI thread entirely. The options stuff I'm looking at specifically because we have a few other bugs happening right now.
